src/execution.jl

```diff
@@ -29,15 +29,15 @@ end
 # execution #
 #############
 # Note that trials executed via `run` and `lineartrial` are always executed at the top-level
-# scope of the module returned by `current_module()`. This is to avoid any weird quirks
+# scope of the module returned by `@__MODULE__`. This is to avoid any weird quirks
 # when calling the experiment from within other contexts.
 
 sample(b::Benchmark, args...) = error("no execution method defined on type $(typeof(b))")
 _run(b::Benchmark, args...; kwargs...) = error("no execution method defined on type $(typeof(b))")
 
 # return (Trial, result) tuple, where result is the result of the benchmarked expression
 function run_result(b::Benchmark, p::Parameters = b.params; kwargs...)
-    return eval(current_module(), :(BenchmarkTools._run($(b), $(p); $(kwargs...))))
+    return eval(@__MODULE__, :(BenchmarkTools._run($(b), $(p); $(kwargs...))))
```

Comment:
Ah, good catch - I haven't been following the `@__MODULE__` changes, I had naively assumed it would expand into an equivalent of `current_module()`. 

To be explicit: `current_module` was being used here to grab the module from which this code was being executed. An example of the intended behavior:

```julia
julia> f() = current_module()
f (generic function with 1 method)

julia> module M
           g() = Main.f()
           println("This should print `M`, not `Main`: ", g())
       end;
This should print `M`, not `Main`: M

julia> module M2
           using M
           g2() = M.g()
           println("This should print `M2`, not `M`: ", g2())
       end;
This should print `M2`, not `M`: M2
```